msal/src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -238,7 +239,7 @@ protected AuthenticationResult CacheTokenResponseAndCreateAuthenticationResult(M
                 fromServer = ClientInfo.CreateFromJson(msalTokenResponse.ClientInfo);
             }
 
-            if (fromServer!= null && AuthenticationRequestParameters?.Account?.HomeAccountId != null)
+            if (fromServer!= null && AuthenticationRequestParameters?.Account?.HomeAccountId != null && AuthenticationRequestParameters.Authority.AuthorityType != AuthorityType.B2C)
```

Comment:
`fromServer.UniqueObjectIdentifier` returns `"b9f....b2c_1_profileeditpolicy"`
and 
`AuthRequestParameters.Account.HomeObjectId` is `b9f...b2c_1_sisopolicy"` 
Using fiddler, we get a token back, so I added a check for b2c authorityType to skip this check and not get the `"Returned user identifiers do not match the sent user identifier"` error. 
Will look into this more next week.

Reply:
Does the tenant id match - can we preserve that verification at least? 

In any case, this validation code is getting complex, please move it to a separate private method.